src/CarbonAware.DataSources/CarbonAware.DataSources.WattTime/test/Configuration/WattTimeClientConfigurationTests.cs

```diff
@@ -27,6 +28,9 @@ public void Validate_DoesNotThrow(string? username, string? password, string? ur
     [TestCase(null, "12345", "http://example.com", TestName = "Validate throws: no username; password; url")]
     [TestCase("testuser", null, "http://example.com", TestName = "Validate throws: no username; password; url")]
     [TestCase(null, null, "http://example.com", TestName = "Validate throws: no username; no password; url")]
+    [TestCase("\ud800", null, "http://example.com", TestName = "Non utf8 characters for username")]
+    [TestCase(null, "\ud83e\udd70", "http://example.com", TestName = "Non utf8 characters for password")]
```

Comment:
```suggestion
    [TestCase("username", "\ud800", "http://example.com", TestName = "Non utf8 characters for password")]
```
Just a small change here. Sorry, I miss guided in the 1st comment.